types_from_serializers/lib/types_from_serializers/generator.rb

```diff
@@ -64,7 +69,13 @@ def ts_properties
                   multi: options[:association] == :many,
                   column_name: options.fetch(:value_from),
                 ).tap do |property|
-                  property.infer_type_from(model_columns, types_from)
+                  enums = model.try(:defined_enums)
+
+                  if enums
+                    enum_values = enums[key].to_h.keys
```

Comment:
Using `key` here means that we would lose type inference when using the alias feature (`tempo: {as: :beat}`).

Adding a commit with a fix, and slight refactoring, otherwise looks great, thanks! 😃